### PR TITLE
Angular5 example

### PR DESCRIPTION
**Pull request by akryvomaz**

### CODE REVIEW

**Comment by kof** [comment]
Thanks for the PR, we need to use a hashtag in commit messages, was it "goodnesssquad" or "goodnesssquadkyiv"?

File: angular5/app/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import { Component, OnInit } from '@angular/core'
+
+import jss from 'jss'
```
Comment (kof) [inline angular5/app/app.component.ts:3]: I would add jss-preset-default so that you can use camel case and other stuff

**Comment by kof** [comment]
Btw. would you like to create and maintain an integration for angular, which can do things react-jss also does?

```
- Theming support out of the box.
- Critical CSS extraction.
- Lazy evaluation - sheet is created only when component will mount.
- Auto attach/detach - sheet will be rendered to the DOM when component is about to mount and will be removed when no element needs it.
- A Style Sheet gets shared between all elements.
```

File: angular5/app/app.component.ts (hunk 1 above)
Comment (akryvomaz) [reply angular5/app/app.component.ts:3]: It is imported, in the app.module.ts

**Comment by akryvomaz** [comment]
> Thanks for the PR, we need to use a hashtag in commit messages, was it "goodnesssquad" or "goodnesssquadkyiv"?

#GoodnessSquadKyiv

> Btw. would you like to create and maintain an integration for angular, which can do things react-jss also does?

Sounds worthy!
I saw this PR - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21851, nice. It is a must before I start with Angular integration. So I can help with the types for TypeScript as well.

File: angular5/app/app.module.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import preset from 'jss-preset-default'
+
+@NgModule({
+  imports: [BrowserModule],
+  declarations: [AppComponent],
+  bootstrap: [AppComponent]
+})
+export class AppModule {
+  constructor() {
+    jss.setup(preset())
```
Comment (kof) [inline angular5/app/app.module.ts]: better to do this statically, outside of a constructor, because it needs to be done only once

**Comment by kof** [comment]
alright, lets finish this pr!

File: angular5/app/app.component.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  public onBlueChanged: Function
+
+  private readonly red: string = '#F44336'
+  private readonly green: string = '#4CAF50'
+  private readonly blue: string = '#2196F3'
+
+  public ngOnInit(): void {
+    const styles: Object = {
+      title: {
+        'text-align': 'center',
```
Comment (kof) [inline angular5/app/app.component.ts]: oh, this can be camel cased, just saw it

File: angular5/app/app.component.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      },
+      area: {
+        width: '100%',
+        height: '10rem',
+        color: 'white',
+        backgroundColor: data => data.area.backgroundColor
+      },
+      buttons: {
+        display: 'flex',
+        'flex-direction': 'row',
```
Comment (kof) [inline angular5/app/app.component.ts]: same here, for consistency

**Comment by kof** [comment]
Merged!

**Pull request merged**